test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,95 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new ConsoleEnvironment();
+            string dataPath = "iris.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, ctx => (
+               label: ctx.LoadText(0),
+               sepalWidth: ctx.LoadFloat(1),
+               sepalLength: ctx.LoadFloat(2),
+               petalWidth: ctx.LoadFloat(3),
+               petalLength: ctx.LoadFloat(4)),
+               dataPath);
```

Comment:
It would be nice if we can make `dataPath` the second parameter here. It is easily missed given the `ctx` is quite involved. @Zruty0 mentioned this is because it is an optional parameter but just thought I'd share in case there are any ideas.

Reply:
I think it's OK to actually have a version of the ctor, where `dataPath` is a required 2nd argument...

---
In reply to: [215140669](https://github.com/dotnet/machinelearning/pull/809#discussion_r215140669) [](ancestors = 215140669)